### PR TITLE
add logic to fix null error for nepnep

### LINKED ISSUE #6987: [Bug] <Mangalife>: <chapters not loading when clicked for 'I Was a Bottom-Tier Bureaucrat for 1,500 Years, and the Demon King Made Me a Minister' and instead an error message comes up>
## Device information
- Tachiyomi version: 0.10.12
- Android version: 10

## Source information
- Name: MangaLife
- Extension version: 1.2.19

## Steps to reproduce
1. Find the manga
2. Click on a chapter

### Expected behavior
I should be able to read the Manga by clicking the chapter

### Actual behavior
The chapter is attempted to be loaded but then stops gets sent back to the chapter list and an error then pop up at the bottom
The error is 
Attempt to invoke virtual method 'java.lang.stringorg.jsoup.nodes.element.data()' on null object reference

## Other details
The manga's name is in the title but I'll put its name here as well ' I Was a Bottom-Tier Bureaucrat for 1,500 Years, and the Demon King Made Me a Minister '
I can access the manga through webview just fine.
I can access other mangas on mangalife without using webview only this one seems to be playing up
[picture of the error that comes up](The error message https://imgur.com/a/qIRxc8o)

### LINKED ISSUE #9982: [Mangasee] Updates failed. Error: java.lang.String org.jsoup.nodes.Element.data()
### Source information

MangaSee 1.2.26

### Steps to reproduce

Update all manga

### Expected behavior

Expect to see new updates if available or nothing when no new update

### Actual behavior

Received this notification: "2 Update failed"

With this as content:
```
Attempt to invoke virtual method 'java.lang.String org.jsoup.nodes.Element.data()' on a null object reference 
# MangaSee (EN) 
- Reincarnated as an Aristocrat with an Appraisal Skill 
- The Unsuccessful yet Academically Unparalleled Sage
```

### Tachiyomi version

0.12.3

### Android version

Android 12

### Other details

Device: Samsung S21 Ultra

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.12.3](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by Seishirou101**
Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [x] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

adds logic to catch null error and do another call to find a page that may have the mainfunction. From the ones I know that give the error, it seems to work but its hard to completely test as there aren't a lot of the error that happens and it usually fixes itself after time. Tested with #6987 as that one still errors but #9982 seems to have been already fixed normally. 

closes #9982
closes #6987

### CODE REVIEW

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/nepnep/NepNep.kt [hunk 1]
```diff
@@ -303,7 +302,14 @@ abstract class NepNep(
 
     override fun pageListParse(response: Response): List<Page> {
         val document = response.asJsoup()
-        val script = document.select("script:containsData(MainFunction)").first().data()
+        val script = try {
+            document.select("script:containsData(MainFunction)").first().data()
+        } catch (e: java.lang.NullPointerException) {
```
Comment (ObserverOfTime) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/nepnep/NepNep.kt:307]:
```suggestion
        val script = document.select("script:containsData(MainFunction)")?.first()?.data()?.let {
```

Comment (Seishirou101) [reply multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/nepnep/NepNep.kt:307]: this doesn't work but I may have done it incorrectly, I have this 
```
val script = document.select("script:containsData(MainFunction)")?.first()?.data()?.let {
            val new_mangaUrl = document.location().removeSuffix(".html")
            val new_response = client.newCall(GET(new_mangaUrl, headers)).execute()
            val new_document = new_response.asJsoup()
            new_document.select("script:containsData(MainFunction)").first().data()
        }

Comment (ObserverOfTime) [reply multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/nepnep/NepNep.kt:307]: My bad, this should work:
```kotlin
        val script = document.selectFirst("script:containsData(MainFunction)")?.data()
            ?: client.newCall(GET(document.location().removeSuffix(".html"), headers))
                .execute().asJsoup().selectFirst("script:containsData(MainFunction)").data()
```

Comment (Seishirou101) [reply multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/nepnep/NepNep.kt]: Thanks! I had to add !! to curChapter for it to compile but it seems everything works!

**Pull request merged**